ElementX/Sources/Mocks/KnockedRoomProxyMock.swift

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright 2024 New Vector Ltd.
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+// Please see LICENSE in the repository root for full details.
+//
+
+import Combine
+import Foundation
+
+@MainActor
+struct KnockedRoomProxyMockConfiguration {
+    var id = UUID().uuidString
+    var name: String?
+    var avatarURL: URL?
+    var members: [RoomMemberProxyMock] = .allMembers
+}
+
+extension KnockedRoomProxyMock {
+    @MainActor
+    convenience init(_ configuration: KnockedRoomProxyMockConfiguration) {
+        self.init()
+        id = configuration.id
+        name = configuration.name
+        avatarURL = avatarURL
```

Comment:
SonarCloud is upset with you!
```suggestion
        avatarURL = configuration.avatarURL
```